.github/workflows/osx.yml

```diff
@@ -0,0 +1,38 @@
+name: github-OSX
+
+on: [push, pull_request]
+
+jobs:
+  osxci:
+    name: osx-ci
+    runs-on: [macos-latest]
+
+    strategy:
+      matrix:
+        include:
+          - BACKEND: "SERIAL"
+            BUILD_TYPE: "RelWithDebInfo"
+          - BACKEND: "PTHREAD"
+            BUILD_TYPE: "RelWithDebInfo"
+          - BACKEND: "SERIAL"
+            BUILD_TYPE: "Debug"
+          - BACKEND: "SERIAL"
+            BUILD_TYPE: "Release"
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: configure
+        run:
+          cmake -B build \
+            -DKokkos_ENABLE_${{ matrix.BACKEND }}=On \
+            -DCMAKE_CXX_FLAGS="-Werror" \
+            -DCMAKE_CXX_STANDARD=14 \
+            -DKokkos_ENABLE_COMPILER_WARNINGS=ON \
+            -DKokkos_ENABLE_TESTS=On \
+            -DCMAKE_BUILD_TYPE=${{ matrix.CMAKE_BUILD_TYPE }} \
```

Comment:
```suggestion
            -DCMAKE_BUILD_TYPE=${{ matrix.BUILD_TYPE }} \
```

Reply:
In #3598 we called the same property `cmake_build_type`, I don't have a strong opinion, but we should make it consistent.

Reply:
I renamed to `cmake_build_type` I don't really care either.